Rules/UseConsistentIndentation.cs

```diff
@@ -254,6 +254,21 @@ public override IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string file
             return diagnosticRecords;
         }
 
+        private static bool PipelineIsFollowedByNewlineOrLineContinuation(Token[] tokens, int tokenIndex)
+        {
+            if (tokenIndex == tokens.Length - 1)
+            {
+                return false;
+            }
+            var nextToken = tokens[tokenIndex + 1];
+            if (nextToken.Kind == TokenKind.Comment && tokenIndex < tokens.Length - 2)
+            {
+                nextToken = tokens[tokenIndex + 2];
+            }
+            return nextToken.Kind == TokenKind.NewLine ||
+                   nextToken.Kind == TokenKind.LineContinuation;
+        }
```

Comment:
Rather than special casing a peek at the next token, I'd just skip all the comments:

```suggestion
        private static bool PipelineIsFollowedByNewlineOrLineContinuation(Token[] tokens, int startIndex)
        {
            if (startIndex >= tokens.Length - 1)
            {
                return false;
            }
            
            Token nextToken = null;
            for (int i = startIndex + 1; i < tokens.Length; i++)
            {
                nextToken = tokens[i];
                
                switch (nextToken.Kind)
                {
                    case TokenKind.Comment:
                        continue;
                        
                    case TokenKind.NewLine:
                    case TokenKind.LineContinuation:
                        return true;
                        
                    default:
                        return false;
            }
            
            // We've run out of tokens but haven't seen a newline
            return false;
        }
```

Reply:
Yh, good point about having multiple comment tokens in there. Thanks for providing the code for it as well, looks more readable as well now :-)